### PR TITLE
Add ProjectSelectorDialogFragmentTest

### LINKED ISSUE #577: [Code health] Create UI test for selecting a project
ProjectDialogTest
===
Create UI tests based on the [happy path](https://github.com/google/ground-android/issues/572). This class should test the following behaviour:

**projectDialogIsShown**
Given: a user who has multiple projects
When: they log in for the first time
Then: the project dialog is displayed and contains a list of all their projects

**selectingAProjectSetsItAsDefault**
Given: a user who has multiple projects and has logged in for the first time 
When: they select a project in the project dialog
Then: that project is set in the user's preferences

**defaultProjectIsShown**
Given: a user who has a default project set in their preferences
When: they login
Then: that project is shown (the project dialog is not shown)

### PR DESCRIPTION
**Pull request by JSunde**
Fixes #577

Adds tests for ProjectSelectorDialog

@gino-m @shobhitagarwal1612 PTAL

### CODE REVIEW

File: gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.android.controller.ActivityController;
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner.class)
+@UninstallModules({LocalDataStoreModule.class})
+public class ProjectSelectorDialogFragmentTest extends BaseHiltTest {
+
+  @Inject ProjectSelectorViewModel viewModel;
```
Comment (shobhitagarwal1612) [inline gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java:58]: This is unused and can be removed.

**Review by shobhitagarwal1612 — COMMENTED** [review]
Some minor nits for your consideration. Looks good overall!

Thanks!

File: gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  private Project project2 = FakeData.newProject().setId("2").build();
+
+  private List<Project> projects = ImmutableList.of(project1, project2);
+
+  @Before
+  public void setup() {
+    fakeRemoteDataStore.setTestProjects(projects);
+//    for (Project project : projects) {
+//      localDataStore.insertOrUpdateProject(project);
+//    }
```
[the remark this answers is not in the document]
Comment (JSunde) [reply gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java:75]: Oops, removed!

File: gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java (hunk 1 above)
Comment (JSunde) [reply gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java]: Done

File: gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  @Inject ProjectRepository projectRepository;
+  @Inject FakeRemoteDataStore fakeRemoteDataStore;
+  @BindValue @Mock LocalDataStore mockLocalDataStore;
+
+  private ProjectSelectorDialogFragment projectSelectorDialogFragment;
+
+  private Project project1 = FakeData.newProject().setId("1").build();
+  private Project project2 = FakeData.newProject().setId("2").build();
+
+  private List<Project> projects = ImmutableList.of(project1, project2);
```
[the remark this answers is not in the document]
Comment (JSunde) [reply gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java]: Done

**Comment by JSunde** [comment]
Huh, it seems strange that the ProjectSelctorDialogFragment is showing 0 coverage in codecov. Any idea why that would be @gino-m?

https://app.codecov.io/gh/google/ground-android/compare/1096/tree/gnd/com/google/android/gnd/ui/projectselector

File: gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+  private ProjectSelectorDialogFragment projectSelectorDialogFragment;
+
+  private final Project project1 = FakeData.newProject().setId("1").build();
+  private final Project project2 = FakeData.newProject().setId("2").build();
+
+  private final List<Project> projects = ImmutableList.of(project1, project2);
+
+  @Before
+  public void setup() {
```
Comment (gino-m) [inline gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java:70]: Nit: s/setup/setUp/ here and below (verb phrase vs noun).

**Comment by gino-m** [comment]
> Huh, it seems strange that the ProjectSelctorDialogFragment is showing 0 coverage in codecov. Any idea why that would be @gino-m?
> 
> https://app.codecov.io/gh/google/ground-android/compare/1096/tree/gnd/com/google/android/gnd/ui/projectselector

Not sure, @shobhitagarwal1612 any idea why this class shows up as 0% coverage? If we're explicitly ignoring these classes I assume they would show up in coverage at all, right?

**Comment by shobhitagarwal1612** [comment]
I don't think it was explicitly ignored. [This](https://github.com/google/ground-android/blob/master/config/jacoco/jacoco.gradle#L28) is where we define it.

Let's file a bug to take care of this separately.

File: gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java (hunk 4 above)
Comment (JSunde) [reply gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java]: Done

**Review by gino-m — APPROVED** [review]
LGTM!

**Review by shobhitagarwal1612 — DISMISSED** [review]

**Comment by shobhitagarwal1612** [comment]
Please fix

```
gnd/src/test/java/com/google/android/gnd/ui/projectselector/ProjectSelectorDialogFragmentTest.java:70: error: [MissingSuperCall] This method overrides com.google.android.gnd.BaseHiltTest#setUp, which is annotated with @OverridingMethodsMustInvokeSuper, but does not call the super method
  public void setUp() {
              ^
    (see https://errorprone.info/bugpattern/MissingSuperCall)
```

**Review by shobhitagarwal1612 — APPROVED** [review]

**Pull request merged**